Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2603 @@
+                currentExtentEndTimeLabel.isHidden = true
+            }
+            else if endTimeLabelX + endTimeLabelSize.width > bounds.maxX - labelSidePadding {
+                endTimeLabelX = bounds.maxX - endTimeLabelSize.width - labelSidePadding
+                currentExtentEndTimeLabel.isHidden = true
+            }
+            else if endTimeLabelX < currentExtentStartTimeLabel.frame.origin.x + currentExtentStartTimeLabel.frame.width {
+                endTimeLabelX = currentExtentStartTimeLabel.frame.origin.x + currentExtentStartTimeLabel.frame.width + paddingBetweenLabels
+            }
+            
+            let thumbEndTimeLabelY = upperThumbLayer.frame.minY - currentExtentEndTimeLabel.frame.height - labelPadding
+            let tickLayerEndTimeLabelY = tickMarkLayer.frame.minY - currentExtentEndTimeLabel.frame.height - labelPadding
+            let endTimeLabelY = min(thumbEndTimeLabelY, tickLayerEndTimeLabelY)
+            currentExtentEndTimeLabel.frame = CGRect(x: endTimeLabelX, y: endTimeLabelY, width: endTimeLabelSize.width, height: endTimeLabelSize.height)
+        }
+        else {
+            currentExtentEndTimeLabel.string = ""
+            currentExtentEndTimeLabel.isHidden = true
+        }
+        
+        // Commit the transaction
+        CATransaction.commit()
+    }
+    
+    private func positionTickMarks() {
+        //
+        // Bail out if time steps are not available
+        guard let timeSteps = timeSteps, !timeSteps.isEmpty, isSliderVisible else {
+            return
+        }
+        
+        // Create tick marks based on
+        // time steps and it's calculated
+        // position on the slider
+        let tms = timeSteps.map { timeStep -> TickMark in
+            let tickX = CGFloat(position(for: timeStep.timeIntervalSince1970))
+            return TickMark(originX: tickX, value: timeStep)
+        }
+        
+        // If label mode is ticks then we need to
+        // find out the major/minor ticks and
+        // add labels for major ticks
+        if labelMode == .ticks {
+            var majorTickInterval = 2
+            var doMajorTickLabelsCollide = false
+            var firstMajorTickIndex = 0
+            let tickCount = tms.count
+            
+            // Calculate the largest number of ticks to allow between major ticks. This prevents scenarios where
+            // there are two major ticks placed undesirably close to the end.
+            let maxMajorTickInterval = Int(ceil(Double(tickCount / 2)))
+            
+            // If maxMajorTickInterval is not greater than majorTickInterval
+            // then we can not do further calculation
+            if maxMajorTickInterval >= majorTickInterval {
+                //
+                // Calculate the number of ticks between each major tick and the index of the first major tick
+                for i in majorTickInterval..<maxMajorTickInterval  {
+                    let prospectiveInterv
```

Comment:
@njarecha While this works, you could let UIKit do the work for you:
1. Specify that the image is a template image in the Asset Catalogue. <img width="266" alt="screen shot 2018-05-15 at 4 57 51 pm" src="https://user-images.githubusercontent.com/2257493/40089605-5f00d5d6-5861-11e8-8034-90b41dfc50a4.png">
2. Set the button's image using `setImage(_:for:)` rather than `setBackgroundImage(_:for:)`.
3. Set the button's `tintColor` to tint the image to the desired color.

Reply:
fixed